src/nomad_parser_fhiaims/parsers/parser.py

```diff
@@ -0,0 +1,195 @@
+                {'name': 'MGGA_C_TPSS'},
+                {'name': 'MGGA_X_TPSS'},
+            ],
+            'TPSSloc gradient-corrected functionals': [
+                {'name': 'MGGA_C_TPSSLOC'},
+                {'name': 'MGGA_X_TPSS'},
+            ],
+            'hybrid B3LYP functional': [{'name': 'HYB_GGA_XC_B3LYP5'}],
+            'Hartree-Fock': [{'name': 'HF_X'}],
+            'HSE': [{'name': 'HYB_GGA_XC_HSE03'}],
+            'HSE-functional': [{'name': 'HYB_GGA_XC_HSE06'}],
+            'hybrid-PBE0 functionals': [
+                {'name': 'GGA_C_PBE'},
+                {
+                    'name': 'GGA_X_PBE',
+                    'weight': lambda x: 0.75 if x is None else 1.0 - x,
+                },
+                {'name': 'HF_X', 'weight': lambda x: 0.25 if x is None else x},
+            ],
+            'hybrid-PBEsol0 functionals': [
+                {'name': 'GGA_C_PBE_SOL'},
+                {
+                    'name': 'GGA_X_PBE_SOL',
+                    'weight': lambda x: 0.75 if x is None else 1.0 - x,
+                },
+                {'name': 'HF_X', 'weight': lambda x: 0.25 if x is None else x},
+            ],
+            'Hybrid M06 gradient-corrected functionals': [
+                {'name': 'MGGA_C_M06'},
+                {'name': 'HYB_MGGA_X_M06'},
+            ],
+            'Hybrid M06-2X gradient-corrected functionals': [
+                {'name': 'MGGA_C_M06_2X'},
+                {'name': 'HYB_MGGA_X_M06'},
+            ],
+            'Hybrid M06-HF gradient-corrected functionals': [
+                {'name': 'MGGA_C_M06_HF'},
+                {'name': 'HYB_MGGA_X_M06'},
+            ],
+            'Hybrid M08-HX gradient-corrected functionals': [
+                {'name': 'MGGA_C_M08_HX'},
+                {'name': 'HYB_MGGA_X_M08_HX'},
+            ],
+            'Hybrid M08-SO gradient-corrected functionals': [
+                {'name': 'MGGA_C_M08_SO'},
+                {'name': 'HYB_MGGA_X_M08_SO'},
+            ],
+            'Hybrid M11 gradient-corrected functionals': [
+                {'name': 'MGGA_C_M11'},
+                {'name': 'HYB_MGGA_X_M11'},
+            ],
+        }
+
+    def get_xc_functionals(self, xc: str) -> List[Dict[str, Any]]:
+        return [dict(name=functional.get('name')) for functional in self._xc_map.get(xc, [])]
+
+    def get_energies(self, source: Dict[str, Any]) -> Dict[str, Any]:
+        total_keys = ['Total energy uncorrected', 'Total energy']
+        energies = {}
+        contributions = []
+        for key, val in source.get('energy', {}).items():
+            if key in total_keys:
+                energies.setdefault('value', val)
+            else:
+                contributions.append({'name': key, 'value': val})
+        for key, val in source.get('energy_components', [{}])[-1].items():
+            contributions.append({'name': key, 'value': val})
+        energies['contributions'] = contributions
+        return energies
```

Comment:
no this to handle the case where the same section def is used for different sub-sections. In this case, TotalEnergy.contributions uses the same def as in TotalEnergy. Since we can define only the def for 'value' once, we have to sort of transform the data so that both TotalEnergy.value and TotalEnergy.contributions.value have similar form.

Reply:
Thank you for clarifying. Pls add this explanation as a pydocs.
I would also use `energies['_contributions']` instead of `energies['contributions']`, to _suggest_ that this name is protected.

Reply:
this is not protected at all, one can use any key as long as it matches the arg for the mapper. it is actually better to use another key to rid of any confusion, will change to `types`

Reply:
> this is not protected at all

Maybe I didn't formulate my comment well. I just meant using the notation, reminiscent of how Python protects object / class variables. As a hint to other devs. This was used in the past by `TextParser`, e.g. the `_results` key.

Am fine with your suggested name change.

Reply:
i am not really understanding it, this is just the transformed source data (dict) which the section parses.

Reply:
I would just use `_components` as the key, that's all. I leave it up to you, no need to keep discussing this.